designs/provisioner-priority.md

```diff
@@ -0,0 +1,268 @@
+# Prioritizing Provisioners
+
+## Goals
+
+- Allowing a user to describe a logical ordering to their provisioners
+- Allowing a user to set defaults and/or fallback orderings for their provisioners
+
+## Background
+
+Currently, there is no logical ordering to how the Karpenter scheduling receives and processes Provisioners. Provisioners are processed in the order they are received from the Kubernetes LIST API, which is randomly. Like pod affinities, users want to convey a preferential relationship to some Provisioners over others. Rather than having to manually map a pod to a specific Provisioner through a `nodeSelector` or `affinity` (which could be cumbersome for clusters with large workloads), a user should be able to specify global preferences to pod assignment to provisioners.
+
+## Use Cases
+
+1. Creating a strongly preferred default provisioner that will always be attempted first, except when more specific configuration is required. This enables scenarios like specifying `kubernetes.io/arch=arm64` for workloads that require this architecture, while preferring `kubernetes.io/arch=amd64` on all other workloads, even if the pod does not contain specific `nodeSelector` constraints. This is particularly useful for those that have high numbers of workloads that only work on `amd64` instances, but they do not want to have to go and assign the specific architecture-specific `nodeSelector` to each of these workloads.
+
+   **Example**
+
+    ```yaml
+    # Default provisioner that will be attempted to be scheduled first
+
+    apiVersion: karpenter.sh/v1alpha5
+    kind: Provisioner
+    metadata:
+      name: default
+    spec:
+      weight: 100
+      requirements:
+      - key: kubernetes.io/arch
+        operator: In
+        values: ["amd64"]
+    ```
+
+    ```yaml
+    # ARM-64 specific provisioner that will be scheduled if the other provisioner does not fit the constraints for this provisioner
+
+    apiVersion: karpenter.sh/v1alpha5
+    kind: Provisioner
+    metadata:
+      name: arm64
+    spec:
+      weight: 50
+      requirements:
+      - key: kubernetes.io/arch
+        operator: In
+        values: ["arm64"]
```

Comment:
Even better, you could just be flexible w/ this one using `Exists`

Reply:
In the case that we have `Exists`, we are saying that the pod requirement must have this label or that the node must have this label? If it's the former, makes sense. If it's the latter, not sure how this helps the scenario that we are suggested